model/src/main/java/org/hibernate/performance/search/model/application/DomainDataUpdater.java

```diff
@@ -139,16 +137,29 @@ public void updateQuestionsAndAnswers(Session session, int invocation, int quest
 		}
 
 		List<QuestionnaireInstance> instances = new EmployeeRepository( session ).findByDefinition( definition );
-		for ( QuestionnaireInstance instance : instances ) {
-			for ( ClosedAnswer answer : instance.getClosedAnswers() ) {
+		for ( QuestionnaireInstance instance : choose( instances, 2 ) ) {
+			for ( ClosedAnswer answer : choose( instance.getClosedAnswers(), 1 ) ) {
```

Comment:
Here's it's fine, since I assume there aren't hundreds of answers in a single questionnaire instance.